packages/nx/src/hasher/hasher.ts

```diff
@@ -361,6 +361,48 @@ class TaskHasher {
     };
   }
 
+  private traverseExternalNodesDependencies(
```

Comment:
Given that the number of external dependencies can be large and we traverse them for every task, should we cache the hash for every external dependency?

Reply:
Yes, I think that would be a good idea. To hash it during the parse phase.

Reply:
I added hash caching in the externalNodes